src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -322,9 +323,10 @@ protected virtual Parameter[] BuildMethodParameters(IReadOnlyDictionary<InputPar
                         // This method has a flattened body
                         if (bodyRequestParameter.Kind == InputOperationParameterKind.Flattened && library != null)
                         {
-                            var objectType = bodyRequestParameter.Type switch
+                            (InputType inputType, bool isNullable) = bodyRequestParameter.Type is InputNullableType nullableType ? (nullableType.Type, true) : (bodyRequestParameter.Type, false);
+                            var objectType = inputType switch
                             {
-                                InputModelType inputModelType => library.ResolveModel(inputModelType).Implementation as SerializableObjectType,
+                                InputModelType inputModelType => library.ResolveModel(inputModelType, isNullable).Implementation as SerializableObjectType,
                                 CodeModelType codeModelType => throw new InvalidOperationException("Expecting model"),
                                 _ => null
                             };
```

Comment:
So `CodeModelType` should have been removed (Wei is on it now)
Therefore the whole purpose of this piece of code is to assert the created typeprovider is SerializableObjectType, we should do it in this way:
```csharp
var csharpType = _typeFactory.CreateType(bodyRequestParameter.Type);
if (csharpType is not {IsFrameworkType: false, Implementation: SerializableObjectType objectType })
{
    throw new InvalidOperationException("Unexpected flattened type");
}
```

Reply:
I cannot left the comment on the part below - the above snippet already includes the below throw.